src/Tribe/Ajax/Dropdown.php

```diff
@@ -95,13 +95,62 @@ public function search_terms( $search, $page, $args, $source ) {
 	}
 
 	/**
-	 * Sorts Hierarchically all the Terms for Select2
+	 * Search for Posts using Select2
+	 *
+	 * @since  TBD
+	 *
+	 * @param  string|array        $search   Search string from Select2.
+	 * @param  int                 $page     When we deal with pagination.
+	 * @param  array,string,mixed> $args     Arguments to pass to the query.
+	 * @param  string|int          $selected Selected item ID.
+	 *
+	 * @return array
+	 */
+	public function search_posts( $search, $page = 1, $args, $selected ) {
+		$posts_per_page = 10;
+		$data           = [];
+
+		if ( ! empty( $search ) ) {
+			if ( is_array( $search ) ) {
+				// Newer SelectWoo uses a new search format.
+				$args['s'] = $search['term']; // post?
+			} else {
+				// For older pieces that still use Select2 format.
+				$args['s'] = $search;
+			}
+		}
+
+		$args['posts_per_page'] = $posts_per_page;
+		$args['paged']          = $page;
+
+		$results = new WP_Query( $args );
+
+		if ( ! $results->have_posts() ) {
+			return $data;
+		}
+
+		foreach( $results->posts as $post ) {
+			// Prep for Select2
+			$post->id        = $post->ID;
```

Comment:
What fields of the post object are **actually** being used in the JS?
If those among to the the post ID and title, then the full post object fetch could be skipped to use a direct query to fetch only the post ID and title.

Not required for this PR, but it's a 1% gain it's worth exploring.

Reply:
To be clear, this would mean running a query to fetch only the post IDs (fast) and then a second one to only fetch the post titles on the matches.

Reply:
Yes - I was duplicating some existing code but I like the simpler approach. Let me check that we're not using the rest of the post and reply.

Reply:
Changed